src/java.base/share/classes/java/lang/Class.java

```diff
@@ -4435,7 +4435,12 @@ public String descriptorString() {
      * Returns a {@code Class} for an array type whose component type
      * is described by this {@linkplain Class}.
      *
+     * @throws IllegalArgumentException if this component type is {@linkplain
+     *         Void#TYPE void} or if the number of dimensions of the resulting array
+     *         type would exceed 255.
```

Comment:
This is a no-arg method so I'm not sure that IAE is the right exception. The method was added in Java 12 so there probably isn't a huge body of code that expects it to throw IAE, meaning it may be possible to change it with little/no impact.